### ISSUE TITLE
[BUG] HTTP2采集的性能问题

### ISSUE THREAD

**Issue by mickeyzzc** [deepflowio/deepflow#4111]
### Search before asking

- [X] I had searched in the [issues](https://github.com/deepflowio/deepflow/issues?q=is%3Aissue) and found no similar feature requirement.

### DeepFlow Component

Agent

### What you expected to happen

![75db7f1a7e67469f52fc96f4a4e258f](https://github.com/deepflowio/deepflow/assets/1990362/6977f178-3a1b-407b-aebe-287fd2100812)
腾讯云的集群，启动http2和关闭http2的采集解码的前后性能差异极大。

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/deepflowio/deepflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by nrjatyunshan** [comment]
> "启动http2和关闭http2的采集解码"

对应的具体操作是什么?可以提供对应开启和关闭状态的采集器配置
@mickeyzzc

**Comment by jiekun** [comment]
> > "启动http2和关闭http2的采集解码"
> 
> 对应的具体操作是什么?可以提供对应开启和关闭状态的采集器配置 @mickeyzzc

这里指的应该是通过 agent group config 控制 agent `l7-protocol-enabled`:
https://github.com/deepflowio/deepflow/blob/main/server/controller/model/agent_group_config_example.yaml#L946C4-L946C24
```yaml
  ## List of Application Protocols
  ## Note: Turning off some protocol identification can reduce deepflow-agent resource consumption.
  ##   ProtobufRPC is disabled by default, note that ebpf-kprobe-whitelist must be configured to enable it.
  #l7-protocol-enabled:
  #- HTTP ## for both HTTP and HTTP_TLS
  #- HTTP2 ## for HTTP2, HTTP2_TLS and gRPC
  #- ProtobufRPC ## for krpc ..
  #- SofaRPC
  #- FastCGI
  #- Dubbo
  #- MySQL
  #- PostgreSQL
  #- Redis
  #- Kafka
  #- MQTT
  #- DNS
  #- Custom ## custom protocol from plugin
```

**Comment by jiekun** [comment]
Any updates? @nrjatyunshan

**Comment by nrjatyunshan** [comment]
@jiekun @mickeyzzc 如果你们两个提到的问题都是由修改 l7-protocol-enabled 引入的,需要 Assignees 中的其他人处理,我对这部分不太了解

**Comment by jiekun** [comment]
ping @rvql @qq906907952 for help.
After enabling HTTP2 protocol support, the performance is affected seriously

**Comment by sharang** [comment]
@rvql, please verify the differences in CPU and MEM resource consumption between HTTP and HTTP2 in v6.4, and address any related performance issues.

**Comment by someview** [comment]
> After enabling HTTP2 protocol support, the performance is affected seriously

Is some progress about this

**Comment by sharang** [comment]
We have optimized the parsing performance of HTTP2, see #5330 for details, enjoy!

@someview @jiekun @mickeyzzc

**Comment by sharang** [comment]
> We have optimized the parsing performance of HTTP2, see #5330 for details, enjoy!
> 
> @someview @jiekun @mickeyzzc

This PR has also been merged into the v6.4 branch.
